lms/djangoapps/discussion_api/api.py

```diff
@@ -583,7 +595,9 @@ def create_comment(request, comment_data):
         raise ValidationError({"thread_id": ["This field is required."]})
     try:
         cc_thread, context = _get_thread_and_context(request, thread_id)
-    except Http404:
+    except CourseNotFoundError:
+        raise ValidationError({"course_id": ["Invalid value."]})
+    except ThreadNotFoundError:
         raise ValidationError({"thread_id": ["Invalid value."]})
```

Comment:
I agree Not found error should not be translated to Validation errors.
This was implemented as is; even test cases assert for validation errors. Let me change this too as part of update.